sonar-application/src/test/java/org/sonar/application/ClusterTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.application;
+
+import com.hazelcast.core.HazelcastException;
+import java.io.IOException;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.NetworkInterface;
+import java.net.ServerSocket;
+import java.net.SocketException;
+import java.util.Enumeration;
+import java.util.Properties;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.sonar.process.Props;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ClusterTest {
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Test
+  public void test_cluster() {
+    Props props = initDefaultProps();
+
+    assertThat(
+      new Cluster(props).isActive()
+    ).isEqualTo(true);
+
+    props.set(ClusterProperties.ENABLED, "false");
+    assertThat(
+      new Cluster(props).isActive()
+    ).isEqualTo(false);
+  }
+
+  @Test
+  public void test_interface() throws SocketException {
+    Props props = initDefaultProps();
+    String ipAddress = findIPv4Address().getHostAddress();
+
+    if (ipAddress != null) {
+      props.set(ClusterProperties.INTERFACES, ipAddress);
+      props.set(ClusterProperties.PORT, "64000");
```

Comment:
hardcoded port means that if SQ UTs are run on the same host concurrently, they may fail because the port is already in use

avoid this. Try and look through existing UTs to find how this is already handled (I don't have a name to provide right now)